libflux/src/flux/semantic/builtins.rs

```diff
@@ -275,6 +275,267 @@ pub fn builtins() -> Builtins<'static> {
+                        ?unit: duration,
+                        ?timeColumn: string,
+                        ?column: string
+                    ) -> [t1]
+                "#),
+                "join" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: t0,
+                        ?method: string,
+                        ?on: [string]
+                    ) -> [t1]
+                "#),
+                // This function would almost have input/output types that match, but:
+                // input column may start as int, uint or float, and always ends up as float.
+                "kaufmansAMA" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        n: int,
+                        ?column: string
+                    ) -> [t1]
+                "#),
+                // either column list or predicate must be provided
+                "keep" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        ?columns: [string],
+                        ?fn: (column: string) -> bool
+                    ) -> [t1]
+                "#),
+                // the "fn" parameter to keyValues is suspicous: https://github.com/influxdata/flux/issues/2235
+                "keyValues" => Node::Builtin(r#"
+                    forall [t0, t1, t2, t3] where t0: Row, t3: Row (
+                        <-tables: [t0],
+                        ?keyColumns: [string],
+                        ?fn: t1
+                    ) -> [{_key: string | _value: t2 | t3}]
+                "#),
+                "keys" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        ?column: string
+                    ) -> [t1]
+                "#),
+                "last" => Node::Builtin("forall [t0] where t0: Row (<-tables: [t0], ?column: string) -> [t0]"),
+                "length" => Node::Builtin("forall [t0] (arr: [t0]) -> int"),
+                "limit"  => Node::Builtin("forall [t0] (<-tables: [t0], n: int, ?offset: int) -> [t0]"),
+                "linearBins" => Node::Builtin(r#"
+                    forall [] (
+                        start: float,
+                        width: float,
+                        count: int,
+                        ?infinity: bool
+                    ) -> [float]
+                "#),
+                "logarithmicBins" => Node::Builtin(r#"
+                    forall [] (
+                        start: float,
+                        factor: float,
+                        count: int,
+                        ?infinity: bool
+                    ) -> [float]
+                "#),
+                "map" => Node::Builtin("forall [t0, t1] (<-tables: [t0], fn: (r: t0) -> t1, ?mergeKey: bool) -> [t1]"),
```

Comment:
I noticed you're not constraining the type variables to be `Row`s. Is that intentional?

Reply:
Yes.  In cases where we have a `column` or `columns` parameter, I left in the `Row` constraint.  But in this case we just have a `mergeKey` parameter, so I was on the fence about it.  It seems like moving forward we'll want to use this one on more than just table streams.

We can remove the `mergeKey` parameter, right?  Since we can now use `with` syntax, there's no reason to use it anymore I think.

Reply:
Good catch. Yes we can remove the mergeKey parameter.

Reply:
Oddly, it looks like it was already removed back in June but has reappeared somehow
https://github.com/influxdata/flux/pull/1416/files#diff-7abc02a1cd9fb8d4c37207d081429b3e

Reply:
I see. It looks like the transpiler uses it. Perhaps it's safe to remove right now, but I'm not sure. I'm fine waiting to remove it when we fix the transpiler.

Reply:
Ok, I reopened the issue, and will link it here.